android/clientlib-ktx/src/main/java/com/solana/mobilewalletadapter/clientlib/ActivityResultSender.kt

```diff
@@ -1,7 +1,35 @@
 package com.solana.mobilewalletadapter.clientlib
 
 import android.content.Intent
+import androidx.activity.ComponentActivity
+import androidx.activity.result.ActivityResultLauncher
+import androidx.activity.result.contract.ActivityResultContracts
+import androidx.annotation.GuardedBy
 
-interface ActivityResultSender {
-    fun launch(intent: Intent)
+class ActivityResultSender(
```

Comment:
Consumers will no longer need to implement this as an interface, but instantiate it, providing their app's component activity.

Reply:
I am not against this change necessarily, but this will break current implementations right? would there be any value to leaving the interface, and providing the implantation here? That would likely be more confusing in the long run but worth discussing?

Reply:
It only changes the implementation _slightly_ for consumers. They used to have to implement the interface and pass it along. This would have been done in their Activity. Now they just instantiate and pass along. 

We don't have any consumers other than ourselves right now, so I'd rather make the breaking change now than later.

Reply:
True true, I dont think its a bad breaking change to make at all. Tho I wonder if we should provide some way for the consumer to provide their own implementation. Do you think this ActivityResultSender implementation will cover all users needs? Maybe it will, but if we think it may need to be tweaked or extended for certain use cases then worth making it an interface with a provided impl IMO.

Reply:
That's a very good question, and probably something we should talk with @sdlaver about.

Reply:
We just started publishing to Maven, and we haven't actively promoted this library yet, so there'll never be a better time to make a breaking change.